output/transform/util.go

```diff
@@ -61,8 +62,20 @@ func upsertQueryReferenceAndInformation(s *snapshot.FullSnapshot, statementTexts
 	return idx
 }
 
-func upsertQueryReferenceAndInformationSimple(server *state.Server, refs []*snapshot.QueryReference, infos []*snapshot.QueryInformation, roleIdx int32, databaseIdx int32, originalQuery string, trackActivityQuerySize int) (int32, []*snapshot.QueryReference, []*snapshot.QueryInformation) {
-	fingerprint := util.FingerprintQuery(originalQuery, server.Config.FilterQueryText, trackActivityQuerySize)
+func upsertQueryReferenceAndInformationSimple(server *state.Server, refs []*snapshot.QueryReference, infos []*snapshot.QueryInformation, roleIdx int32, databaseIdx int32, queryID null.Int, originalQuery string, trackActivityQuerySize int) (int32, []*snapshot.QueryReference, []*snapshot.QueryInformation) {
+	var fingerprint uint64
+	var fingerprintFound bool
+	if server.QueryIdentities != nil && queryID.Valid {
```

Comment:
I think in order to read `server.QueryIdentities` here without holding the mutex we'd have to ensure it is set outside of a mutex (i.e. when a server struct is created) - right now this could theoretically be a bad pointer read (though maybe setting a pointer is an atomic operation in Go?), e.g. when an activity snapshot calls this whilst a full snapshot just starts the GetStatements loop.

Reply:
I'll update the code so the identity map always exists, so this nil check isn't needed.